src/System.Windows.Forms/src/System/Windows/Forms/Control.cs

```diff
@@ -2099,8 +2099,72 @@ public virtual Font Font
             [param: MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(ActiveXFontMarshaler))]
             set
             {
-                SetFont(value);
+                var local = (Font)Properties.GetObject(s_fontProperty);
+                bool localChanged = false;
+                if (value is null)
+                {
+                    if (local is not null)
+                    {
+                        localChanged = true;
+                    }
+                }
+                else
+                {
+                    localChanged = local is null ? true : !value.Equals(local);
+                }
```

Comment:
This can be simplified to:
```suggestion
                bool localChanged;
                if (value is null)
                {
                    localChanged = local is not null;
                }
                else
                {
                    localChanged = local is null ? true : !value.Equals(local);
                }
```

This can be further inlined, but to me it didn't look any better.